cmd/admin-handlers-users.go

```diff
@@ -521,8 +526,15 @@ func (a adminAPIHandlers) DeleteServiceAccount(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	if cred.AccessKey != user || cred.ParentUser != user {
-		// The service account belongs to another user but return not found error to mitigate brute force attacks.
+	parentUser := cred.AccessKey
+	if cred.ParentUser != "" {
+		parentUser = cred.ParentUser
+	}
+
+	if parentUser != user || user == "" {
```

Comment:
Maybe globalIAMSys.GetServiceAccountParent() can return an error instead of `"", nil` so we can avoid `|| user = ""` here.

Reply:
We don't need to return error for that @vadmeste because non existent accounts are okay to be requested by client. An empty user means that we are definitely sure that we didn't find the user. I added extra code which is not needed because parentUser is always non-empty and the check would work automatically. 

It was another defensive code.